lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeNames.cpp

```diff
@@ -136,6 +136,37 @@ AreFuncletsOfSameAsyncClosure(swift::Demangle::NodePointer closure1,
   return Node::deepEquals(closure1_function, closure2_function);
 }
 
+std::optional<uint64_t>
+SwiftLanguageRuntime::GetFuncletNumber(swift::Demangle::NodePointer node) {
+  if (!node)
+    return {};
+  NodePointer node_id =
+      childAtPath(node, {Node::Kind::AsyncSuspendResumePartialFunction,
+                         Node::Kind::Number});
+  if (!node_id)
+    node_id = childAtPath(node, {Node::Kind::AsyncAwaitResumePartialFunction,
+                                 Node::Kind::Number});
+  if (node_id)
+    return node_id->getIndex();
+
+  // If this is an async function and there was no number, then this is an entry
+  // funclet. Assign it number 0.
+  if (SwiftLanguageRuntime::IsAnySwiftAsyncFunctionSymbol(node))
+    return 0;
+
+  return {};
+}
+
+std::optional<uint64_t>
+SwiftLanguageRuntime::GetFuncletNumber(llvm::StringRef name) {
+  if (!IsSwiftMangledName(name))
+    return {};
+  using namespace swift::Demangle;
+  Context ctx;
+  NodePointer node = SwiftLanguageRuntime::DemangleSymbolAsNode(name, ctx);
+  return GetFuncletNumber(node);
```

Comment:
Should there be a log (perhaps verbose only) to capture symbols that fail to get a funclet number?

```suggestion
  auto number = GetFuncletNumber(node);
  if (node && !number)
    LLDB_LOG(..., "could not get funclet number for mangled name {0}", name);
  return number;
```

Reply:
The challenge with this is that the `StringRef` overload of this function is only used for testing.

The real users of this function will use the `NodePointer` overload (which is called by the StringRef overload), as they use the `NodePointer` for other purposes. As such, we wouldn't be logging on the most important path (there is no way to grab the name from the NodePointer).

With that in mind, I'll make sure to add logging on the next PR, the one that uses these functions to filter breakpoints.